plugins/languages/otter.nix

```diff
@@ -0,0 +1,90 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "otter";
+  originalName = "otter.nvim";
+  defaultPackage = pkgs.vimPlugins.otter-nvim;
+
+  maintainers = [ lib.maintainers.perchun ];
+
+  settingsOptions = {
+    lsp = {
+      hover = {
+        border = helpers.defaultNullOpts.mkListOf lib.types.str [
+          "╭"
+          "─"
+          "╮"
+          "│"
+          "╯"
+          "─"
+          "╰"
+          "│"
+        ] "";
+      };
+
+      diagnostic_update_events = helpers.defaultNullOpts.mkListOf' {
+        type = lib.types.str;
+        pluginDefault = [ "BufWritePost" ];
+        description = ''
+          `:h events` that cause the diagnostics to update.
+
+          See example for less performant but more instant diagnostic updates.
+        '';
+        example = [
+          "BufWritePost"
+          "InsertLeave"
+          "TextChanged"
+        ];
+      };
+    };
+
+    buffers = {
+      set_filetype = helpers.defaultNullOpts.mkBool false ''
+        If set to true, the filetype of the otterbuffers will be set.
+        Otherwise only the autocommand of lspconfig that attaches
+        the language server will be executed without setting the filetype.
+      '';
+
+      write_to_disk = helpers.defaultNullOpts.mkBool false ''
+        Write `<path>.otter.<embedded language extension>` files
+        to disk on save of main buffer.
+        Useful for some linters that require actual files,
+        otter files are deleted on quit or main buffer close.
+      '';
+    };
+
+    strip_wrapping_quote_characters = helpers.defaultNullOpts.mkListOf lib.types.str [
+      "'"
+      "\""
+      "\`"
+    ] "";
+
+    handle_leading_whitespace = helpers.defaultNullOpts.mkBool false ''
+      Otter may not work the way you expect when entire code blocks are indented (eg. in Org files).
+      When true, otter handles these cases fully. This is a (minor) performance hit.
+    '';
+  };
+
+  extraOptions = {
+    addCmpSources = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
```

Comment:
I think normally this would be default false, and then we'd have an assertion for `cfg.addCmpSources -> config.plugins.cmp.enable`.

Although most users probably want this, we're trying to prefer explicit opt-in config over having automation and "magic". For completion sources in particular, users may want to set a priority, or add it to a different sources list.

Reply:
So should I set it default to false? I would expect it automatically be added to cmp sources (because this is one of the main features of otter), and then if I want it to tweak, disable the option and add it manually.

Reply:
I think it's fine in this instance